datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -685,7 +685,8 @@ impl OptimizerRule for PushDownFilter {
                         .map(LogicalPlan::Filter)?;
                 insert_below(LogicalPlan::Repartition(repartition), new_filter)
             }
-            LogicalPlan::Distinct(distinct) => {
+            LogicalPlan::Distinct(distinct @ Distinct::All(_)) => {
+                // note that we check for distinct all as distinct on is not commutable
```

Comment:
Unless the filter is only on the distinct columns?

Reply:
> Do you know any tool or function that can help with that?

This function https://github.com/apache/datafusion/blob/8acd5cb0f13620c85d86d9ebdb8475a3905f7714/datafusion/core/src/datasource/listing/helpers.rs#L54 is used for determining if an Expr can be evaluated using a subset of columns (currently use for pushdown in to the table scan). Perhaps it could be used?